### PR TITLE
Update versions to latest CP 6.2.1 (AK 2.8.1)

### LINKED ISSUE #869: Update the Kafka Streams branching tutorial
[KIP-418](https://cwiki.apache.org/confluence/display/KAFKA/KIP-418%3A+A+method-chaining+way+to+branch+KStream) introduced a new way to perform branching on a Kafka Stream topology and it deprecated the previous approach.  The current Kafka Streams branching tutorial uses the deprecated approach, and we should update it.

### LINKED ISSUE #948: Update KT to 6.2.1/AK2.8.1

### PR DESCRIPTION
**Pull request by ybyzek**
### Description

https://github.com/confluentinc/kafka-tutorials/issues/948

A lot of files were updated in this PR:

- `docker-compose.yml` and `build.gradle` files: versions updated (CP 6.2.1, AK 2.8.1, Voluble 0.3.0)
- Some individual tutorials needed code updates as well, due to deprecated or removed methods, or changing ksqlDB syntax: see specific links in section below which draw attention to the modified tutorials (those that had changes beyond just version number update)

### Staging Docs

Landing page:
- http://kafka-tutorials-staging.s3-website-us-west-2.amazonaws.com/GH-948

Individual tutorials that were modified beyond version updates (these may warrant closer inspection):
- http://kafka-tutorials-staging.s3-website-us-west-2.amazonaws.com/GH-948/split-a-stream-of-events-into-substreams/kstreams.html
- http://kafka-tutorials-staging.s3-website-us-west-2.amazonaws.com/GH-948/kafka-console-consumer-primitive-keys-values/kafka.html
- http://kafka-tutorials-staging.s3-website-us-west-2.amazonaws.com/GH-948/change-topic-partitions-replicas/ksql.html
- http://kafka-tutorials-staging.s3-website-us-west-2.amazonaws.com/GH-948/generate-streams-of-test-data/ksql.html
- http://kafka-tutorials-staging.s3-website-us-west-2.amazonaws.com/GH-948/anomaly-detection/ksql.html
- http://kafka-tutorials-staging.s3-website-us-west-2.amazonaws.com/GH-948/split-a-stream-of-events-into-substreams/ksql.html
- http://kafka-tutorials-staging.s3-website-us-west-2.amazonaws.com/GH-948/transform-a-stream-of-events/ksql.html
- http://kafka-tutorials-staging.s3-website-us-west-2.amazonaws.com/GH-948/produce-consume-lang/scala.html
- http://kafka-tutorials-staging.s3-website-us-west-2.amazonaws.com/GH-948/connect-add-key-to-source/ksql.html

### CODE REVIEW

**Comment by ybyzek** [comment]
@rspurgeon @bbejeck moving back to Draft status to resolve failing test, please hold off on reviews

File: _data/harnesses/generate-test-data-streams/ksql.yml
Comment (rspurgeon) [inline _data/harnesses/generate-test-data-streams/ksql.yml:191]: Could this be (realtively easily) replaced with some kind of deterministic check similar to the existing `wait-for-containers.sh` script? The goal being toavoid the pitfalls of sleeps:
1. trying to find and manage over time the optimal wait time
2. making the user wait unecessarily

Comment (ybyzek) [reply _data/harnesses/generate-test-data-streams/ksql.yml:191]: The user doesn't experience this wait, it's only in Semaphore, and it's not the long pole.  IMO it's not worth the extra cycles to figure out the automated check for this one.

File: _includes/contribute-tutorial-lang.html
Comment (rspurgeon) [inline _includes/contribute-tutorial-lang.html:7]: I've had "here" links flagged a bunch in the past, FYC:
```suggestion
            <div class="title">A tutorial for this language is available on <a href="https://developer.confluent.io/get-started">Confluent Developer</a></div>
```

Comment (ybyzek) [reply _includes/contribute-tutorial-lang.html:7]: I almost wrote `Confluent Developer` but KT will be part of CD very shortly, so it would refer to itself

File: _includes/tutorials/splitting/kstreams/markup/dev/make-topology.adoc
Comment (rspurgeon) [inline _includes/tutorials/splitting/kstreams/markup/dev/make-topology.adoc:1]: I think this change resulted in some incorrect language:

* The `split` function is overloaded to take no arguments or an instance of `Named` (https://kafka.apache.org/28/javadoc/org/apache/kafka/streams/kstream/KStream.html#split())
* `Branched` -> `branch` and it takes the predicate function as the parameter

File: _includes/contribute-tutorial-lang.html
Comment (rspurgeon) [reply _includes/contribute-tutorial-lang.html:7]: I still suggest some language besides "here".  Additionally it might be nice if we linked the user directly to the specific language page as with this link they land on a banner that says Java and that might be enough for them to back away

Comment (ybyzek) [reply _includes/contribute-tutorial-lang.html:7]: Updated in latest commit

File: _includes/tutorials/splitting/kstreams/markup/dev/make-topology.adoc
Comment (ybyzek) [reply _includes/tutorials/splitting/kstreams/markup/dev/make-topology.adoc]: Updated in latest commit

**Review by bbejeck — COMMENTED** [review]
Thanks for PR @ybyzek, overall looks good.  I just have one minor comment.  

I didn't go through and validate anything individually, instead, I'm thinking we should wait to merge until we get a green build.

File: _includes/tutorials/error-handling/kstreams/code/build.gradle
Comment (ybyzek) [reply _includes/tutorials/error-handling/kstreams/code/build.gradle:37]: 👍

File: _includes/tutorials/cogrouping-streams/kstreams/code/build.gradle
Comment (ybyzek) [reply _includes/tutorials/cogrouping-streams/kstreams/code/build.gradle]: 👍 updated in latest commit

**Comment by ybyzek** [comment]
@bbejeck we have an all green build ;). Please review/approve when you have a chance, thanks!

**Review by bbejeck — APPROVED** [review]
LGTM! Stellar work @ybyzek!

**Pull request merged**
